classes/Builder/Payload/OnboardingPayloadBuilder.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+* 2007-2019 PrestaShop
+*
+* NOTICE OF LICENSE
+*
+* This source file is subject to the Academic Free License (AFL 3.0)
+* that is bundled with this package in the file LICENSE.txt.
+* It is also available through the world-wide-web at this URL:
+* http://opensource.org/licenses/afl-3.0.php
+* If you did not receive a copy of the license and are unable to
+* obtain it through the world-wide-web, please send an email
+* to license@prestashop.com so we can send you a copy immediately.
+*
+* DISCLAIMER
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+*  @author    PrestaShop SA <contact@prestashop.com>
+*  @copyright 2007-2019 PrestaShop SA
+*  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+*  International Registered Trademark & Property of PrestaShop SA
+*/
+
+namespace PrestaShop\Module\PrestashopCheckout\Builder\Payload;
+
+use PrestaShop\Module\PrestashopCheckout\PsxDataMatrice\PsxDataMatrice;
+use PrestaShop\Module\PrestashopCheckout\Repository\PsAccountRepository;
+
+/**
+ * Build the payload for getting paypal onboarding link
+ */
+class OnboardingPayloadBuilder implements PayloadBuilderInterface
+{
+    /**
+     * @var Payload
+     */
+    private $payload;
+
+    public function __construct()
+    {
+        $this->reset();
+    }
+
+    public function reset()
+    {
+        $this->payload = new Payload();
+    }
+
+    /**
+     * Return the result of the payload
+     * Clean the builder to be ready to producing a new payload
+     *
+     * @return Payload
+     */
+    public function getPayload()
```

Comment:
This method is not only a getter as it resets the Patload attribute. I recommend using another verb, like `purgePayload` or `getAndResetPayload`